src/main/java/com/gooddata/GoodDataEndpoint.java

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (C) 2004-2016, GoodData(R) Corporation. All rights reserved.
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE.txt file in the root directory of this source tree.
+ */
+package com.gooddata;
+
+import org.apache.http.HttpHost;
+
+import static com.gooddata.util.Validate.notEmpty;
+
+/**
+ * GoodData Platform endpoint represented by host, port and protocol
+ */
+public class GoodDataEndpoint {
+
+    public static final String PROTOCOL = "https";
+    public static final int PORT = 443;
+    public static final String HOSTNAME = "secure.gooddata.com";
+
+    private final String hostname;
+    private final int port;
+    private final String protocol;
+
+    /**
+     * Create GoodData endpoint for given hostname, port and protocol
+     * @param hostname  GoodData Platform's host name (e.g. secure.gooddata.com)
+     * @param port      GoodData Platform's API port (e.g. 443)
+     * @param protocol  GoodData Platform's API protocol (e.g. https)
+     */
+    public GoodDataEndpoint(final String hostname, final int port, final String protocol) {
+        this.hostname = notEmpty(hostname, "hostname");
+        this.port = port;
```

Comment:
consider to validate the range (yes I know http client will validate it later, but it could be convenient to validate everything at one place?)